### DISCUSSION TITLE
Cannot connect to openhab

### DISCUSSION THREAD

**Discussion by MarinusPretorius** [openhab/openhab-google-assistant#419] [General]
I have setup everything as per this document.  But when I try to link the test openhab in google home it keeps on telling me that it cannot connect please try again.  Any help please?

**Comment by michikrug** [comment]
Could you please add more information on your setup as there are many steps involved?

What does "i have setup everything as per this document" mean?

**Reply by MarinusPretorius** [reply]
Hi,

Followed the steps as per this page

https://github.com/openhab/openhab-google-assistant

Regards
Marinus

**Reply by michikrug** [reply]
I got you. But again without specifying any details on your setup, we can not help you.

Did you look into logs of all the included components to find errors?

**Reply by MarinusPretorius** [reply]
I am not sure what info you are looking for

I have aws ec2 instance running my openhab cloud. Everything working

I want to connect google home to that openhab instance. That is why I have downloaded this code. Build the function and deployed to google cloud. Created an action to connect to function.

The function log gives no info. Just that it triggered the function with http 200 response

In google home i cluck add device and select [test] openhab. It redirects to my instance of openhab. I login and allow auth to google home. Then it shows that its setting up my account. Then after a while it just fails with message that it could not connect to openhab.

Hope this gives you better idea

**Reply by michikrug** [reply]
Alright. Thanks. That helps indeed.

Do you already have openhab items tagged with metadata according to the documentation?

Does the log of the cloud function list a SYNC command?

Edit: woops you said it does not list anything.

But the function needs to be called with at least some intent... 

It should look something like:
<img width="883" alt="Bildschirm­foto 2022-12-01 um 23 06 58" src="https://user-images.githubusercontent.com/727517/205169337-0c843bfa-d068-4e1d-89e3-6b26bb0e962d.png">

**Reply by MarinusPretorius** [reply]
Hi,

Here is from my logs.  This fires every couple of minutes

2022-12-02 06:49:50.089 SAST
openhabGoogleAssistantmyts0gvl4e0d Function execution started
2022-12-02 06:49:50.094 SAST
openhabGoogleAssistantmyts0gvl4e0d openhabGoogleAssistant - onSync
2022-12-02 06:49:50.103 SAST
openhabGoogleAssistantmyts0gvl4e0d Function execution took 14 ms. Finished with status code: 200

Not sure if this is maybe something.  But when I try to test the function from google clouds functions dashboard I get this

2022-12-02 08:21:46.241 SAST
openhabGoogleAssistant9ig0bxn5ovsr Function execution started
2022-12-02 08:21:48.741 SAST
openhabGoogleAssistant9ig0bxn5ovsr TypeError: Cannot read property '0' of undefined at Function.handler (/workspace/node_modules/actions-on-google/dist/service/smarthome/smarthome.js:165:39) at standard (/workspace/node_modules/actions-on-google/dist/assistant.js:48:32)
2022-12-02 08:21:48.741 SAST
openhabGoogleAssistant9ig0bxn5ovsr at /workspace/node_modules/actions-on-google/dist/framework/express.js:28:13
2022-12-02 08:21:48.741 SAST
openhabGoogleAssistant9ig0bxn5ovsr at omni (/workspace/node_modules/actions-on-google/dist/assistant.js:37:53)
2022-12-02 08:21:48.741 SAST
openhabGoogleAssistant9ig0bxn5ovsr at /layers/google.nodejs.functions-framework/functions-framework/node_modules/@google-cloud/functions-framework/build/src/function_wrappers.js:101:17
2022-12-02 08:21:48.742 SAST
openhabGoogleAssistant9ig0bxn5ovsr at processTicksAndRejections (internal/process/task_queues.js:77:11)
2022-12-02 08:21:48.749 SAST
openhabGoogleAssistant9ig0bxn5ovsr Function execution took 2507 ms. Finished with status code: 500

Regards
Marinus

**Reply by michikrug** [reply]
OK. So the SYNC is happening. But as I guessed, it does not find any devices at your openHAB to expose to Google. This is very likely the cause that the linking does not succeed. Can you check your item configuration?

**Reply by MarinusPretorius** [reply]
My openhab on my phone is connected but it does not show in my private myopenhab cloud under devices. Is this what you are talking about?

Any tips how to resolve this maybe?

**Reply by michikrug** [reply]
Nope. Devices is not relevant but your instance should be listen in "online".

Regarding the config I was referring to the actual item configuration in you openHAB. So adding the GA metadata in the proper way so that items are exposed to Google.

**Reply by MarinusPretorius** [reply]
Oh Yes, that is all correct because I use the same items in the myopenhab.org and the live version of Openhab in google home

**Reply by michikrug** [reply]
OK so you then also changed the openhab cloud endpoint in your openHAB instance to point to your own myopenhab url?

**Reply by MarinusPretorius** [reply]
Yes.

Some more contect. I created a custom ifttt action as well to my own myopenhab. That works 100%

**Reply by michikrug** [reply]
Alright.

Then still my most confusing question is then, why SYNC is not adding any devices but also no error message is logged.
Especially, as you say the official integration works though.

**Reply by MarinusPretorius** [reply]
Strange

Error in google home when trying to link to [test] openhab

Could not reach [test] openhab. Please try again

**Reply by MarinusPretorius** [reply]
This is after the oauth allowed in my private myopenhab

**Reply by MarinusPretorius** [reply]
Appreciate your quick responses

**Reply by MarinusPretorius** [reply]
I had 5 items exposed in openhab to ifttt.  I had a lot of all items that came up in the action list in ifttt.  I went and click on the delete items button in openhab without realizing that I would not be able to get my items reloaded.  Now I am stuck.  

I tried setting up my own openhab cloud, link to ifttt and google home.  But now I am stuck at this linking with google home.

I feel it is such an overkill to go through this whole process just to get my 5 items in ifttt back.

Is there any information maybe of enabling the items syncing in openhab again?

Would it be possible the send a request to add 5 items to my myopenhab.org account?

**Reply by MarinusPretorius** [reply]
OMG, there I got it to work.  I added https in front of my host in the config.js file.  By reading the installation steps again very carefully I found this 

Change host to point to your openHAB Cloud instance, for example: openhab.myserver.com. Do not include https, if you do you'll get DNS errors.

Probably missed it because I never had dns errors.

Thanks again for all you assistance

**Reply by michikrug** [reply]
Whoa. Nice to hear that. Will consider adjusting the code to may strip out the https if given. 👍🏻👍🏻

**Reply by michikrug** [reply]
Just FYI. I now added an error log line for such cases.
